From b98c06e5bfe451a7d88347302d612e9da62f5b0c Mon Sep 17 00:00:00 2001
From: Feng Song <56996751+hisongfeng@users.noreply.github.com>
Date: Fri, 12 Jun 2020 18:27:58 +0900
Subject: [PATCH] add-usr_config.mk-for-in-app-can-enable-disable-config

1, add usr_config.mk
2, solve the conflict issue with ctype.h

Signed-off-by: feng.song <56996751+hisongfeng@users.noreply.github.com>
---
 component/common/api/platform/platform_stdlib_rtl8721d.h | 16 ++++++++--------
 .../GCC-RELEASE/project_hp/asdk/config.mk                |  2 ++
 2 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/component/common/api/platform/platform_stdlib_rtl8721d.h b/component/common/api/platform/platform_stdlib_rtl8721d.h
index 40c75a5..79575ad 100644
--- a/component/common/api/platform/platform_stdlib_rtl8721d.h
+++ b/component/common/api/platform/platform_stdlib_rtl8721d.h
@@ -87,18 +87,18 @@
 	#define rand						Rand
 	#define srand				
 	
+//undef below for conflict with ctype.h
+	#undef in_range
+	#undef isprint
+	#undef isdigit
+	#undef isxdigit
+	#undef islower
+	#undef isspace
+
 	//extern int _sscanf_patch(const char *buf, const char *fmt, ...);
 	//#define sscanf					_sscanf_patch
 	
 
 #endif	// defined (__IARSTDLIB__)
 
-extern void *pvPortMalloc( size_t xWantedSize );
-extern void vPortFree( void *pv );
-#define malloc                  pvPortMalloc
-#define free                    vPortFree
-#define realloc			pvPortReAlloc
-#define calloc			rtw_calloc
-
-
 #endif // PLATFORM_STDLIB_8721D_H
diff --git a/project/realtek_amebaD_va0_example/GCC-RELEASE/project_hp/asdk/config.mk b/project/realtek_amebaD_va0_example/GCC-RELEASE/project_hp/asdk/config.mk
index 21ad22d..eadb830 100644
--- a/project/realtek_amebaD_va0_example/GCC-RELEASE/project_hp/asdk/config.mk
+++ b/project/realtek_amebaD_va0_example/GCC-RELEASE/project_hp/asdk/config.mk
@@ -2,6 +2,8 @@
 # Config
 ########################################
 
+include $(PROJECT_PATH)/usr_config.mk
+
 #/* For STDK */
 CONFIG_STDK_IOT_CORE=y
 CONFIG_STDK_IOT_CORE_USE_MBEDTLS=y
-- 
2.7.4

